src/ConfigWidgets/SymbolsDialog.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ConfigWidgets/SymbolsDialog.h"
+
+#include <qpushbutton.h>
+
+#include <QFileDialog>
+#include <QPushButton>
+#include <algorithm>
+#include <memory>
+
+#include "OrbitBase/Logging.h"
+#include "ui_SymbolsDialog.h"
+
+namespace orbit_config_widgets {
+
+SymbolsDialog::~SymbolsDialog() noexcept = default;
+
+SymbolsDialog::SymbolsDialog(QWidget* parent)
+    : QDialog(parent), ui_(std::make_unique<Ui::SymbolsDialog>()) {
+  ui_->setupUi(this);
+
+  QObject::connect(ui_->addButton, &QPushButton::clicked, this, [this]() {
+    QString directory = QFileDialog::getExistingDirectory(this, "Select Symbol Directory");
+    if (!directory.isEmpty()) {
+      AddSymbolPath(std::filesystem::path{directory.toStdString()});
+    }
+  });
+
+  QObject::connect(ui_->removeButton, &QPushButton::clicked, this,
+                   &SymbolsDialog::OnRemoveButtonClicked);
+
+  QObject::connect(ui_->listWidget, &QListWidget::itemSelectionChanged, this,
+                   &SymbolsDialog::OnSelectionChanged);
+
+  // Initially disabled, will be enabled once selection changes
+  ui_->removeButton->setEnabled(false);
+}
+
+void SymbolsDialog::SetSymbolPaths(const std::vector<std::filesystem::path>& paths) {
```

Comment:
Maybe `absl::Span` instead of `std::vector`?

Reply:
good idea. done